packages/vite/src/node/config.ts

```diff
@@ -1390,6 +1390,21 @@ export async function resolveConfig(
     'ssr.',
   )
 
+  // For backward compat, set ssr environment build.emitAssets with the same value as build.ssrEmitAssets that might be changed in configResolved hook
+  // https://github.com/vikejs/vike/blob/953614cea7b418fcc0309b5c918491889fdec90a/vike/node/plugin/plugins/buildConfig.ts#L67
+  if (resolved.environments.ssr) {
+    resolved.environments.ssr.build.emitAssets =
+      resolved.build.ssrEmitAssets || resolved.build.emitAssets
+  }
+  // For backward compat, set client environment dev.optimizeDeps with the same value as root optimizeDeps that might be changed in configResolved hook
+  // https://github.com/vikejs/vike/blob/953614cea7b418fcc0309b5c918491889fdec90a/vike/node/plugin/plugins/devConfig/index.ts#L70
+  if (resolved.environments.client) {
+    resolved.environments.client.dev.optimizeDeps.include =
+      resolved.optimizeDeps.include
+    resolved.environments.client.dev.optimizeDeps.entries =
+      resolved.optimizeDeps.entries
+  }
```

Comment:
I'm not sure if this assignment is safe though.

Reply:
The `optimizeDeps` part would not be safe. I added a commit to share the object directly so modifications are reflected in both side [chore: backward compatibility using same optimizeDeps object](https://github.com/vitejs/vite/pull/17947/commits/03c02eeeb8f90b45eb178762c4cd90107c21713e). Let me know if this sounds ok.

Reply:
Sounds ok to me 👍